gnd/src/main/java/com/google/android/gnd/persistence/local/room/RoomLocalDataStore.java

```diff
@@ -260,6 +261,8 @@ public Completable applyAndEnqueue(FeatureMutation mutation) {
         .map(
             list ->
                 stream(list)
+                    .filter(
+                        observationEntity -> observationEntity.getState() != EntityState.DELETED)
```

Comment:
Can we implement this in the DAO instead?

Reply:
Thanks for pointing out! Dao already contains a where clause for DEFAULT state